tests/unit/test_endpoint_scan_analysis.py

```diff
@@ -0,0 +1,188 @@
+            mock.add('POST',
+                     url=consts.ANALYZE_URL + '/scans/scans/1234/memory/collected-binaries',
+                     status=HTTPStatus.OK,
+                     json={'result': {'status': 'success'}})
+            mock.add('POST',
+                     url=consts.ANALYZE_URL + '/scans/scans/1234/end',
+                     status=HTTPStatus.OK,
+                     json={'result': {'status': 'success'}})
+
+            # Act
+            analysis.send()
+
+        # Assert
+        self.assertEqual(analysis.status, consts.AnalysisStatusCode.CREATED)
+
+    def test_send_analyze_to_api_missing_files(self):
+        # Arrange
+        metadata_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                     '..',
+                                                     'resources',
+                                                     'offline_endpoint_scan',
+                                                     'metadata_dir_missing_files'))
+        analysis = EndpointAnalysis(metadata_dir=metadata_dir)
+
+        # Arrange
+        with responses.RequestsMock() as mock:
+            mock.add('POST',
+                     url=consts.BASE_URL + 'scans',
+                     status=HTTPStatus.CREATED,
+                     json={'result': {'scan_id': '1234', 'analysis_id': '5678'}})
+            mock.add('POST',
+                     url=consts.ANALYZE_URL + '/scans/scans/1234/host-info',
+                     status=HTTPStatus.OK,
+                     json={'result': {'status': 'success'}})
+            mock.add('POST',
+                     url=consts.ANALYZE_URL + '/scans/scans/1234/end',
+                     status=HTTPStatus.OK,
+                     json={'result': {'status': 'failed'}})
+
+            # Act
+            try:
+                analysis.send()
+            except Exception as e:
+                # Assert
+                error = str(e)
+                self.assertTrue(error.startswith("[Errno 2] No such file or directory:") and
+                                error.endswith("metadata_dir_missing_files/processes_info.json'"))
+
+        self.assertEqual(analysis.status, consts.AnalysisStatusCode.FAILED)
+
+    def test_send_analyze_to_api_no_scheduled_tasks_info(self):
+        # Arrange
+        metadata_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                     '..',
+                                                     'resources',
+                                                     'offline_endpoint_scan',
+                                                     'metadata_dir_no_scheduled_tasks'))
+        analysis = EndpointAnalysis(metadata_dir=metadata_dir)
+
+        # Arrange
+        with responses.RequestsMock() as mock:
+            mock.add('POST',
```

Comment:
I think it will be better if you'll extract the basic calls to functions so we won't have copy-paste in the tests